brew_view/controllers/roles_api.py

```diff
@@ -0,0 +1,209 @@
+            type: string
+        responses:
+          200:
+            description: Role with the given ID
+            schema:
+              $ref: '#/definitions/Role'
+          404:
+            $ref: '#/definitions/404Error'
+          50x:
+            $ref: '#/definitions/50xError'
+        tags:
+          - Roles
+        """
+        self.write(BeerGardenSchemaParser.serialize_role(
+            Role.objects.get(id=str(role_id)),
+            to_string=False
+        ))
+
+    def delete(self, role_id):
+        """
+        ---
+        summary: Delete a specific Role
+        parameters:
+          - name: role_id
+            in: path
+            required: true
+            description: The ID of the Role
+            type: string
+        responses:
+          204:
+            description: Role has been successfully deleted
+          404:
+            $ref: '#/definitions/404Error'
+          50x:
+            $ref: '#/definitions/50xError'
+        tags:
+          - Roles
+        """
+        role = Role.objects.get(id=str(role_id))
+        role.delete()
+
+        self.set_status(204)
+
+    def patch(self, role_id):
+        """
+        ---
+        summary: Partially update a Role
+        description: |
+          The body of the request needs to contain a set of instructions
+          detailing the updates to apply:
+          ```JSON
+          {
+            "operations": [
+              { "operation": "add", "path": "/permissions", "value": "ALL" }
+            ]
+          }
+          ```
+        parameters:
+          - name: role_id
+            in: path
+            required: true
+            description: The ID of the Role
+            type: string
+          - name: patch
+            in: body
+            required: true
+            description: Instructions for how to update the Role
+            schema:
+              $ref: '#/definitions/Patch'
+        responses:
+          200:
+            description: Role with the given ID
+            schema:
+              $ref: '#/definitions/Role'
+          400:
+            $ref: '#/definitions/400Error'
+          404:
+            $ref: '#/definitions/404Error'
+          50x:
+            $ref: '#/definitions/50xError'
+        tags:
+          - Roles
+        """
+        role = Role.objects.get(id=str(role_id))
+        operations = BeerGardenSchemaParser.parse_patch(
+            self.request.decoded_body,
+            many=True,
+            from_string=True
+        )
+
+        for op in operations:
+            if op.path == '/permissions':
+                try:
+                    perm = Permissions(op.value)
+                except ValueError:
+                    error_msg = "Permission '%s' does not exist" % op.value
+                    self.logger.warning(error_msg)
+                    raise ModelValidationError(error_msg)
+
+                if op.operation == 'add':
+                    role.permissions.append(perm.value)
```

Comment:
Do we need to be doing de-duping here?

Same (ish) question for removals. Will removing a value that is not there cause an error?

Reply:
I'm fine if the answer is "yes it will throw a validation error"